### DISCUSSION TITLE
Can Roslyn Analyzers be used to analyze .NET Framework (4.8)?

### DISCUSSION THREAD

**Discussion by jol64** [dotnet/roslyn#55931] [Q&A]
I am writing a Roslyn Analyzer to help my colleagues to squeeze out systematically what I perceive as a bad practice. The analyzer and also some fixes work when configured to be used in a .NET Core library. However I did not succeed to configure a .NET Framework library to get analyzed. I also find questions like the following
https://stackoverflow.com/questions/67634772/using-roslyn-in-a-net-4-8-library-project-reference
https://github.com/login?return_to=%2Fdotnet%2Froslyn%2Fissues%2F50612
I am wondering whether this is actually supported or not, and what options there are to address that. One optio might be to switch to .NET Standard libraries, but that appears to be hard either.
Thanks, Joachim

**Comment by Youssef1313** [comment]
Analyzers are supposed to be TFM-agnostic. Can you clarify what problems do you face?

**Comment by Youssef1313** [comment]
You may also consider starting from <https://github.com/jmarolf/analyzer-start>.

**Comment by jol64** [comment]
> Analyzers are supposed to be TFM-agnostic. Can you clarify what problems do you face?

After using Nuget Manage...  and adding the analyzer, the analyzer dlls are not even shown in the dependencies tree of .NET Framework class library project. For .NET standard and Core the analyzer dlls are listed.
I am also struggeling and spot issues with getting them to run in the other projects. It does work in the analyzer source project, but not when importing via Nuget. Could be some dependency issue, I guess.

**Comment by Youssef1313** [comment]
It may be easier to investigate if you share a repro project, or if you can try start from the linked repo above and see if it's working for you. If it does work, then you can start comparing it with your project.

**Comment by jol64** [comment]
I am not allowed to share source code.

> You may also consider starting from https://github.com/jmarolf/analyzer-start.

I cloned and tried to build. I am getting
![image](https://user-images.githubusercontent.com/24666601/128341227-ae6596c2-1b3b-4d86-8b90-4078b6c2aba9.png)
Any other option?

**Comment by sharwell** [comment]
> Any other option?

You can use the template in Visual Studio:

![image](https://user-images.githubusercontent.com/1408396/128538773-3c48cc8f-6832-46da-b30f-c59d998a5042.png)

**Comment by jol64** [comment]
I used the template to start with. I don´t think I did something off track.
With other option I was more looking for a means to debug why VS doesn´t accept the analyzer - like a log that validates any missing prerequesites. It just silently ignores my command...

**Comment by jol64** [comment]
It appears to work with migrated sdk projects:
http://hermit.no/moving-to-sdk-style-projects-and-package-references-in-visual-studio-part-2/ "dotnet migrate-2019 wizard"...

**Comment by DynConcepts** [comment]
Still looking for a sample 4.8 C# project/solution that will run the built-in Rosyln analyzers.... (which is what the original title of this discussion was). No matter how bad I write code (deliberately) I do not get any warnings [and ideally the ability to treat them as errors].

**Reply by sharwell** [reply]
We would likely need to see a sample project to provide further help on this issue.

**Reply by davidcorbin-atmosera** [reply]
Your wish fulfilled
[CheckingForDisposal.zip](https://github.com/dotnet/roslyn/files/13284243/CheckingForDisposal.zip)
 [thanks for taking a look]. The solution contains 2 projects... .NET 6.0 which should give the warnings on build... and .NET 4.8 which I can not get to generate the warnings no matter if VS-2022 or VS-2019 (16.11.19)..
